src/webgl/p5.RendererGL.Immediate.js

```diff
@@ -197,7 +203,11 @@ p5.RendererGL.prototype._drawFillImmediateMode = function(
       0
     );
   }
-  if (this.drawMode === constants.TEXTURE) {
+
+  if (
+    this.drawMode === constants.TEXTURE &&
+    this.curFillShader.attributes.aTexCoord
+  ) {
```

Comment:
hmm, if the drawMode is TEXTURE but there are no texture coordinates available, this feels like maybe an error worth noting to the user? (under what circumstances is this edge case possible—is this true if we're drawing lines?)

Reply:
i don't think that's an error. you could certainly have a shader that uses a texture, but does not use the aTexCoord vertex attributes.